src/window_identifier/mod.rs

```diff
@@ -310,6 +314,47 @@ impl Serialize for WindowIdentifierType {
     }
 }
 
+impl WindowIdentifierType {
+    /// Sets the given window as a modal child of the window represented by
+    /// self. Currently only Wayland windows are supported.
```

Comment:
You should also add a note that this will only work if a wayland window is placed above another wayland window, not above an x11 one. For that one would need a private wayland protocol from the compositor to do so, mutter does that with `x11_interop` and https://gitlab.gnome.org/GNOME/libgxdp/-/tree/main?ref_type=heads